auth/src/main/java/com/alibaba/nacos/auth/AuthLdapAuthenticationProvider.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 1999-2018 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacos.auth;
+
+import com.alibaba.nacos.auth.roles.AuthNacosRoleServiceImpl;
+import com.alibaba.nacos.auth.roles.RoleInfo;
+import com.alibaba.nacos.auth.users.NacosUserDetails;
+import com.alibaba.nacos.auth.users.AuthNacosUserDetailsServiceImpl;
+import com.alibaba.nacos.auth.users.User;
+import com.alibaba.nacos.auth.util.PasswordEncoderUtil;
+import com.alibaba.nacos.common.utils.CollectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+
+import javax.naming.CommunicationException;
+import javax.naming.Context;
+import javax.naming.directory.DirContext;
+import javax.naming.ldap.InitialLdapContext;
+import javax.naming.ldap.LdapContext;
+import java.util.Hashtable;
+import java.util.List;
+
+import static com.alibaba.nacos.auth.roles.AuthNacosRoleServiceImpl.GLOBAL_ADMIN_ROLE;
+
+/**
+ * LDAP auth provider.
+ *
+ * @author zjw
+ */
+@Component
+public class AuthLdapAuthenticationProvider implements AuthenticationProvider {
```

Comment:
The class name is strange. There are two `Auth` meaning. I think can remove the first one.